### PR TITLE
GEP: Client Certificate Validation for TLS terminating at the Gateway Listener

### PR DESCRIPTION
**Pull request by arkodg**
/kind gep

Fixes https://github.com/kubernetes-sigs/gateway-api/issues/91

### CODE REVIEW

**Comment by arkodg** [comment]
will continue adding more content in the GEP once Ive received broad consensus on the goals & non goals

File: geps/gep-91.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+(See definitions in [GEP Status][/contributing/gep#status].)
+
+## TLDR
+
+This GEP proposes a way to validate the TLS certificate presented by the downstream client to the server
+(Gateway Listener in this case) during a [TLS Handshake Protocol][], also commonly referred to as mutual TLS (mTLS).
+
+## Goals
+- Define an API field to specify the CA Certificate within the Gateway Listener configuration that can be used as a trusted anchor to validate the certificates presented by the client.
```
Comment (frankbu) [inline geps/gep-91.md:14]: Isn't the existing `certificateRefs` field where the CA certificate would be specified? Is the idea that this GEP will add a field that can be used to specify something like a client validation cacert key in a `cacertificateRefs` secret, for example? Maybe adding an API section with a proposal would be helpful to start the discussion?

Comment (frankbu) [reply geps/gep-91.md:14]: fwiw, this goal of adding a way to config mTLS termination in a Gateway is valuable and very contained scope IMO.

Comment (robscott) [reply geps/gep-91.md:14]: Agree that feels like the inevitable placement here. Let's get this in and then we can work on the implementation details.

**Comment by robscott** [comment]
Thanks @arkodg! Will give @shaneutt and @youngnick a bit more time to review this one, but don't think there's anything controversial here.

/approve

**Comment by frankbu** [comment]
@robscott @shaneutt @youngnick 

Hey guys, since this is no more than a simple (I believe agreed to) problem statement that has been stagnant for a couple of months, can we either get this merged now or otherwise update it with an API proposal so we can start discussing the solution?

@argkodg do you have an API for this that you are waiting to propose?

**Comment by robscott** [comment]
Thanks for the ping on this one @frankbu! I don't think there's anything controversial in the GEP at this point. I've already approved, would like @shaneutt or @youngnick to sign off on this though.

/release-note-none

**Comment by youngnick** [comment]
Yeah, this is pretty tightly scoped as written, so I'm supportive.

/lgtm

**Pull request merged**